test/unit/test-postoperator.cpp

```diff
@@ -482,4 +483,143 @@ TEST_CASE("GridFunction export", "[gridfunction][Serial][Parallel]")
+                                                                                scaling);
+  U_m = std::make_unique<EnergyDensityCoefficient<EnergyDensityType::MAGNETIC>>(*B, mat_op,
+                                                                                scaling);
+  S = std::make_unique<PoyntingVectorCoefficient>(*E, *B, mat_op, scaling);
+
+  // Expected scalings.
+  auto mesh_Lc0 = iodata.units.GetMeshLengthRelativeScale();
+  // E [V/m].
+  auto scaling_E = iodata.units.Dimensionalize<Units::ValueType::FIELD_E>(1.0) * mesh_Lc0;
+  // B [Wb/m²].
+  auto scaling_B =
+      iodata.units.Dimensionalize<Units::ValueType::FIELD_B>(1.0) * mesh_Lc0 * mesh_Lc0;
+  // V [V].
+  auto scaling_V = iodata.units.Dimensionalize<Units::ValueType::VOLTAGE>(1.0);
+  // A [A/m].
+  auto scaling_A = iodata.units.Dimensionalize<Units::ValueType::CURRENT>(1.0) * mesh_Lc0;
+  // Ue = 1/2 ε_0 Eᴴ E.
+  auto scaling_Ue = iodata.units.Dimensionalize<Units::ValueType::FIELD_E>(1.0) *
+                    iodata.units.Dimensionalize<Units::ValueType::FIELD_E>(1.0);
+  // U_m = 1/2 μ⁻¹ Bᴴ B.
+  auto scaling_Um = iodata.units.Dimensionalize<Units::ValueType::FIELD_B>(1.0) *
+                    iodata.units.Dimensionalize<Units::ValueType::FIELD_B>(1.0);
+  // S = Re{E x μ⁻¹B⋆}.
+  auto scaling_S = iodata.units.Dimensionalize<Units::ValueType::FIELD_E>(1.0) *
+                   iodata.units.Dimensionalize<Units::ValueType::FIELD_B>(1.0) * mesh_Lc0 *
+                   mesh_Lc0;
+  std::vector<double> expected_scaling = {scaling_E,  scaling_E,  scaling_B,
+                                          scaling_B,  scaling_V,  scaling_A,
+                                          scaling_Ue, scaling_Um, scaling_S};
+  auto get_max_values = [&]()
+  {
+    std::vector<double> values;
+    values.push_back(E->Real().Max());
+    values.push_back(E->Imag().Max());
+    values.push_back(B->Real().Max());
+    values.push_back(B->Imag().Max());
+    values.push_back(V->Real().Max());
+    values.push_back(A->Real().Max());
+    gridfunc_h1 = 0.0;
+    gridfunc_h1.ProjectCoefficient(*U_e.get());
+    values.push_back(gridfunc_h1.Max());
+    gridfunc_h1 = 0.0;
+    gridfunc_h1.ProjectCoefficient(*U_m.get());
+    values.push_back(gridfunc_h1.Max());
+    gridfunc_rt = 0.0;
+    gridfunc_rt.ProjectCoefficient(*S.get());
+    values.push_back(gridfunc_rt.Max());
+    return values;
+  };
+
+  // Get max values before dimensionalizing.
+  std::vector<double> values_before = get_max_values();
+
+  // Dimensionalize.
+  mesh::DimensionalizeMesh(mesh, mesh_Lc0);
+  ScaleGridFunctions(mesh_Lc0, mesh.Dimension(), true, E, B, V, A);
+  DimensionalizeGridFunctions(iodata.units, true, E, B, V, A);
+  std::vector<double> dim_values = get_max_values();
+  // Check values are as expected.
+  for (size_t i = 0; i < dim_values.size(); i++)
+  {
+    CHECK_THAT(dim_values[i], WithinRel(values_before[i] * expected_scaling[i]));
+  }
```

Comment:
I think we could make this test much more stringent by:
- Making a copy of the initial gridfunctions
- Dimensionalizing into new copies
- Taking the ratio between new and old ones
- Checking that the min and the max of the ratio is `expected_scaling`

I understand that in this case it did not matter because the grid function is constant, but I think that being constant makes the test less effective